packages/flutter/lib/src/widgets/routes.dart

```diff
@@ -1367,15 +1339,161 @@ class RouteObserver<R extends Route<dynamic>> extends NavigatorObserver {
 abstract class RouteAware {
   /// Called when the top route has been popped off, and the current route
   /// shows up.
-  void didPopNext() { }
+  void didPopNext() {}
 
   /// Called when the current route has been pushed.
-  void didPush() { }
+  void didPush() {}
 
   /// Called when the current route has been popped off.
-  void didPop() { }
+  void didPop() {}
 
   /// Called when a new route has been pushed, and the current route is no
   /// longer visible.
-  void didPushNext() { }
+  void didPushNext() {}
+}
+
+class _DialogRoute<T> extends PopupRoute<T> {
+  _DialogRoute({
+    @required this.child,
+    bool barrierDismissible = true,
+    String barrierLabel = 'Dismiss',
+    Color barrierColor = const Color(0x80000000),
+    Duration transitionDuration = const Duration(milliseconds: 200),
+    DialogPageBuilder pageBuilder,
+    DialogTransitionBuilder transitionBuilder,
+    RouteSettings settings,
+  })  : assert(barrierDismissible != null),
+        _barrierDismissible = barrierDismissible,
+        _barrierLabel = barrierLabel,
+        _barrierColor = barrierColor,
+        _transitionDuration = transitionDuration,
+        _pageBuilder = pageBuilder,
+        _transitionBuilder = transitionBuilder,
+        super(settings: settings);
+
+  final Widget child;
+
+  @override
+  bool get barrierDismissible => _barrierDismissible;
+  final bool _barrierDismissible;
+
+  @override
+  String get barrierLabel => _barrierLabel;
+  final String _barrierLabel;
+
+  @override
+  Color get barrierColor => _barrierColor;
+  final Color _barrierColor;
+
+  @override
+  Duration get transitionDuration => _transitionDuration;
+  final Duration _transitionDuration;
+
+  final DialogPageBuilder _pageBuilder;
+
+  final DialogTransitionBuilder _transitionBuilder;
+
+  @override
+  Widget buildPage(BuildContext context, Animation<double> animation, Animation<double> secondaryAnimation) {
+    if (_pageBuilder == null) {
```

Comment:
Seems like one can either supply a builder or a child but child was marked `@required` in the constructor.